### PR TITLE
#159965480 Users should be able to create their profiles

### PR DESCRIPTION
**Pull request by DerKip**
## What does this PR do?
```
- Enables users to create and update their profiles and their profiles.
- It also enables other users to view other peoples profile.
```

## Description of Task to be completed?
```
- Implement Profile object creation upon creating a user
- Create a  URL  and its view that a allows a user's profile to be viewed
- Create an exception handler for users not found
- Create a method to generate and Avatar for a user profile
- Write unit  tests for the added features

```
## How should this be manually tested?

- clone the repo 
- checkout ft-profiles-159965480 branch 
- Register a user by sending a POST request to  ```http://127.0.0.1:8000/api/users/```
- Then check whether the user profile associated with it is created by sending a GET request to 
   ```http://127.0.0.1:8000/api/profiles/<username>```
- To edit update user data, which shall be reflected in the profile. Go to  ```http://127.0.0.1:8000/api/user/``` and change the desired fields. 
An example:
```
{
  "user": {
     "bio": "I bake ",
    "image":" https://vsco.co/tichtoo/media/544f672f7367089c0c8b4606"
  }
}
```
## What are the relevant pivotal tracker stories?

[#159965480](https://www.pivotaltracker.com/story/show/159965480)

### CODE REVIEW

**Comment by ghost** [comment]
@DerKip Some checks haven’t completed yet, is there a reason for this ???

File: authors/apps/authentication/signals.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+
+from authors.apps.profiles.models import Profile
+
+from .models import User
+
+@receiver(post_save, sender=User)
+def save_profile(sender, instance, created, **kwargs):
```
Comment (ghost) [inline authors/apps/authentication/signals.py:10]: @DerKip Please add comments for the methods/functions/classes

**Comment by Mutuba** [comment]
Consider using docstrings in classes and functions definitions!

**Review by WNjihia — COMMENTED** [review]
@DerKip Is the user able to edit their profile? I haven't seen functionality for that

**Comment by DerKip** [comment]
@rethes, that has been implemented.. the profile details of a user can be updated by this endpoint ```http://127.0.0.1:8000/api/user``` that is after, authentication.

**Review by ghost — DISMISSED** [review]
@DerKip Your implementation of the task is good.

File: authors/apps/authentication/signals.py (hunk 1 above)
Comment (DerKip) [reply authors/apps/authentication/signals.py:10]: I've implemented the changes @rethes @Mutuba

**Review by Quantum-35 — DISMISSED** [review]

**Review by Mutuba — APPROVED** [review]
l like you added the docstrings.

**Review by ghost — APPROVED** [review]

**Review by WNjihia — APPROVED** [review]

**Pull request merged**